File: backendSpringboot/FreelancerProjet/src/main/java/com/freelancer/model/Proposition.java
package com.freelancer.model;

import java.util.Date;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Lob;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Proposition {
@Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id ;
@Lob
	private String lettreMotivation ;
	private double price;
	private String portfolioLink ;
	private boolean accepter;
	private boolean refuser ;
	private int workIn;
	private boolean enattent;
	private Date sendingDate;
@ManyToOne
@JoinColumn(name = "idProject")
	private Project project;
@ManyToOne
@JoinColumn(name = "idcv")
	private CV cv;

@ManyToOne
@JoinColumn(name = "idclient")
	private Utilisateurs client;
}

File: frontendAngular/FreelancerAdmin/src/app/monClass/proposition.ts
 import { Utilisateur } from "./client";
import { CV } from "./cv";
 import { Project } from "./project";

export class Proposition {
    id:number;
    titre:string
    lettreMotivation:string ;
    price:number
    workIn:number
    sendingDate:string
    portfolioLink:string
    accepter:boolean
    refuser:boolean 
    enattent:boolean
    project:Project
    cv:CV
    client:Utilisateur

}